### PR TITLE
[google_sign_in_web] Migrate to pkg:web.

### PR DESCRIPTION
**Pull request by ditman**
This PR updates `google_sign_in_web` to:

* Use the latest `google_identity_services_web` and `http`
* Use `package:web`

This should allow the package to compile to WASM.

## Issues

* Fixes https://github.com/flutter/flutter/issues/139169
* Needed to https://github.com/flutter/flutter/issues/139170 

## Tests

* All tests should pass (ran locally)
* Demo app compiled to WASM: https://dit-gis-test.web.app

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Review by kevmoo — APPROVED** [review]
LGTM

File: packages/google_sign_in/google_sign_in_web/example/pubspec.yaml [hunk 1]
```diff
@@ -2,8 +2,8 @@ name: google_sign_in_web_integration_tests
 publish_to: none
 
 environment:
-  sdk: ^3.2.0
-  flutter: ^3.16.0
+  sdk: ">=3.2.0 <4.0.0"
```
Comment (kevmoo) [inline packages/google_sign_in/google_sign_in_web/example/pubspec.yaml:5]: `^` should be fine here! Weird, I know...

Comment (ditman) [reply packages/google_sign_in/google_sign_in_web/example/pubspec.yaml:5]: Haha yeah, I noticed that this is a flutter-only thing, however I saw other entries of the sdk like that one so I decided to make them the same to make it easier to find-replace in the future? maybe?

Comment (kevmoo) [reply packages/google_sign_in/google_sign_in_web/example/pubspec.yaml:5]: 🤷 no biggy!

**Comment by ditman** [comment]
This is ready to land.

**Comment by amrgetment** [comment]
@ditman as a support for you
https://github.com/dart-lang/pub-dev/issues/7279

I wonder if there are a document for migration is it only dart:html to pkg:web or what else do we need to migrate'dart:js' too?

I know this is not the place to ask, it is just a quick question

**Comment by kevmoo** [comment]
@amrgetment – we're working on these docs. We'll have them early next year!

**Pull request merged**